src/rt/region/externalreference.h

```diff
@@ -199,11 +150,11 @@ namespace verona::rt
 
     void merge(Alloc* alloc, ExternalReferenceTable* that)
     {
-      for (auto& e : *that->external_map)
+      for (auto e : *that->external_map)

```

Comment:
The iterator is now returning copies of the entries, so the `std::move` does nothing useful?  I think this code is going to perform an incref and decref for everything it merges, rather than no inc/dec refs.

Reply:
the entry type here is `std::pair<Object*, ExternalRef*>` so there won't be any implicit increfs/decrefs with the raw pointers. I'll add a commit soon to forward moved arguments properly

Reply:
If we merge two tables that are completely disjoint then this is going to cause increfs, as the `insert_unique` will call `on_insert` as it wasn't there before, and the `std::move` is not going to invalidate the original version in table being merged from, so the destructor will call `on_delete`?

Could you log the increfs and decrefs before and after your changes?

Reply:
before:
```
theodus|~/src/microsoft/verona/src/rt/build(master !)$ ninja func-con-rememberedset && ./func-con-rememberedset
[2/2] Linking CXX executable func-con-rememberedset
incref 0x603000000130
decref 0x603000000130
incref 0x6030000000a0
decref 0x6030000000a0
decref 0x6030000000a0
decref 0x603000000130
incref 0x6030000002e0
decref 0x6030000002e0
decref 0x603000000250
decref 0x6030000002e0
incref 0x6030000005b0
incref 0x6030000005b0
decref 0x6030000005b0
decref 0x6030000005b0
decref 0x603000000520
decref 0x6030000005b0
decref 0x603000000490
incref 0x603000000820
incref 0x603000000820
decref 0x603000000820
decref 0x603000000820
decref 0x603000000700
decref 0x603000000790
decref 0x603000000820
```

after:
```
theodus|~/src/microsoft/verona/src/rt/build(theo-hashmap !)$ ninja func-con-rememberedset && ./func-con-rememberedset
[2/2] Linking CXX executable func-con-rememberedset
incref 0x603000000130
decref 0x603000000130
incref 0x6030000000a0
decref 0x6030000000a0
decref 0x6030000000a0
decref 0x603000000130
incref 0x6030000002e0
decref 0x6030000002e0
decref 0x603000000250
decref 0x6030000002e0
incref 0x6030000005b0
incref 0x6030000005b0
decref 0x6030000005b0
decref 0x6030000005b0
decref 0x603000000520
decref 0x6030000005b0
decref 0x603000000490
incref 0x603000000820
incref 0x603000000820
decref 0x603000000820
decref 0x603000000820
decref 0x603000000700
decref 0x603000000790
decref 0x603000000820
```

Reply:
Thanks.  Could you run that on the external reference tests?  It is that merge code that worries me.

Reply:
Ah ok, now I see what you mean. This does add infcrefs/decrefs when merging. I'll try to figure out how to fix that

Reply:
I've brought the increfs/decrefs back down to what they were previously. Sorry, I didn't look closely enough at the shenanigans in the wrapper class move constructor.

Reply:
Thanks.  No worries. It was only on second reading that I realised the issue. C++ tricks to make clients harder to get wrong, have implementation hard to get right ;-)